tests/oeedger8r/host/testdeepcopy.cpp

```diff
@@ -185,6 +197,30 @@ void test_deepcopy_edl_ecalls(oe_enclave_t* enclave)
         test_struct(s, 3);
     }
 
+    {
+        auto s = init_struct<CountStruct>();
+        int ints[]{0, 1, 2, 3};
+        ShallowStruct shallow{1, 8, nullptr};
+        CountStruct counts[]{s, s, s};
+        NestedStruct n{13, ints, &shallow, counts};
+        OE_TEST(deepcopy_nested_out(enclave, &n) == OE_OK);
+
+        for (size_t i = 0; i < 4; ++i)
+            OE_TEST(n.array_of_int[i] == static_cast<int>(i));
+
+        // Out-only deepcopy does not support shallow pointer passing.
```

Comment:
Can you elaborate in this?

Reply:
Nested shallow pointer passing relies on deep copying the address of the pointer and passing the address to the enclave (via the input buffer). Therefore, the enclave gets the address points to the pointer on the host side (i.e., pass-by-address). However, for an out-only parameter, the content of the nested structure will be zero-out (using the output buffer instead) and therefore not deep copied to the enclave side.

For example,
`struct ShallowStruct {
    uint64_t count;
    size_t size;
    uint64_t* ptr;
 };`
In the case of `[in, count=1] ShallowStruct* s`, the value of `ptr` will be copied to the enclave.
For the case of `[out, count=1] ShallowStruct* s`, the value of `ptr` will be zero-out.

Hope this makes sense!